src/codeu/chat/client/ClientMessage.java

```diff
@@ -165,11 +171,13 @@ private void showPreviousMessages(int count) {
    * 				Keyword to find in the messages.
    */
   public void findMessages(String keyword){
-	  
-	  for(final Message m : conversationContents){
-		  if(m.content.contains(keyword)){
+	  // Check the first element to determine if it the StoreLink is empty	  
+	  if(messagesSortedByKeyword.first(keyword) != null){
```

Comment:
Could we rather have a member function for Store<K, V> called isEmpty() or size() to check for emptiness? I think that would be more elegant.